hibernate-core/src/main/java/org/hibernate/loader/ast/internal/ExecutionContextWithSubselectFetchHandler.java

```diff
@@ -4,30 +4,49 @@
  */
 package org.hibernate.loader.ast.internal;
 
+import org.hibernate.LockOptions;
 import org.hibernate.engine.spi.EntityHolder;
 import org.hibernate.engine.spi.SharedSessionContractImplementor;
 import org.hibernate.engine.spi.SubselectFetch;
 import org.hibernate.query.spi.QueryOptions;
+import org.hibernate.query.spi.QueryOptionsAdapter;
 import org.hibernate.sql.exec.internal.BaseExecutionContext;
 
 class ExecutionContextWithSubselectFetchHandler extends BaseExecutionContext {
 
 	private final SubselectFetch.RegistrationHandler subSelectFetchableKeysHandler;
 	private final boolean readOnly;
+	private final QueryOptions queryOptions;
 
 	public ExecutionContextWithSubselectFetchHandler(
 			SharedSessionContractImplementor session,
 			SubselectFetch.RegistrationHandler subSelectFetchableKeysHandler) {
-		this( session, subSelectFetchableKeysHandler, false );
+		this( session, subSelectFetchableKeysHandler, false, null );
```

Comment:
Instead of passing `null` to the other constructor, I think this is probably clearer:

```suggestion
		super( session );
		this.subSelectFetchableKeysHandler = subSelectFetchableKeysHandler;
		this.readOnly = readOnly;
		this.queryOptions = QueryOptions.READ_ONLY;
```